prime-router/src/main/kotlin/azure/ApiKeysFunctions.kt

```diff
@@ -0,0 +1,105 @@
+package gov.cdc.prime.router.azure
+
+import com.microsoft.azure.functions.HttpMethod
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import com.microsoft.azure.functions.annotation.AuthorizationLevel
+import com.microsoft.azure.functions.annotation.BindingName
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.HttpTrigger
+import gov.cdc.prime.router.Organization
+import gov.cdc.prime.router.azure.db.enums.SettingType
+import gov.cdc.prime.router.common.BaseEngine
+import gov.cdc.prime.router.common.JacksonMapperUtilities
+import gov.cdc.prime.router.tokens.AuthenticatedClaims
+import gov.cdc.prime.router.tokens.JwkSet
+import gov.cdc.prime.router.tokens.Scope
+import gov.cdc.prime.router.tokens.SenderUtils
+import gov.cdc.prime.router.tokens.authenticationFailure
+import org.apache.logging.log4j.kotlin.Logging
+
+class ApiKeysFunctions(private val settingsFacade: SettingsFacade = SettingsFacade.common) : Logging {
+
+    data class ApiKeysResponse(val orgName: String, val keys: List<JwkSet>)
+
+    @FunctionName("getApiKeys")
+    fun get(
+        @HttpTrigger(
+            name = "getApiKeys",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "settings/organizations/{organizationName}/public-keys"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("organizationName") orgName: String
+    ): HttpResponseMessage {
+        val claims = AuthenticatedClaims.authenticate(request)
+        if (claims == null || !claims.authorized(setOf("*.*.primeadmin", "$orgName.*.admin"))) {
```

Comment:
Is the `"$orgName.*.admin"` part of this correct?  I don't think we have any Organization admins in Okta 🤔

Reply:
Let me confirm this, but the `AuthenticatedClaims` class supports both Okta and Server2Server

Reply:
Confirming that at least from my reading of the code, that this should work
https://github.com/CDCgov/prime-reportstream/blob/2a95aecf703a523634a132315ebac58bd0ff91a3/prime-router/src/main/kotlin/tokens/AuthenticatedClaims.kt#L192